lib/entity_linking.py

```
@@ -157,11 +212,23 @@ def nex_link(text):
     """
+    # Check cache first
+    cache_params = {
+        'min_confidence': NEX_DATA.get('min_confidence', 0.1)
+    }
+    key = get_cache_key(text, 'nex', **cache_params)
+    cached = get_cached_result(key)
+    if cached is not None:
+        return cached
+
     NEX_DATA['text'] = text
     r = requests.post(NEX_URL, data=NEX_DATA)
```

Ruff (0.15.7)

[error] Probable use of `requests` call without timeout

(S113)